File: Part1_강의자료/Ch08_투포인터/문제별코드/14465_소가길을건너간이유5/14465_prefix_sum.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, K, B;
    scanf("%d %d %d", &N, &K, &B);
    
    int b;
    vector<bool> is_broken(N + 1);
    while (B--)
    {
        scanf("%d", &b);
        is_broken[b] = true;
    }

    vector<int> acc_broken(N + 1);
    for (int i = 1; i <= N; i++)
        acc_broken[i] = acc_broken[i - 1] + is_broken[i];
    
    int ans = K, cur;
    for (int i = 1; i <= N - K + 1; i++)
    {
        cur = acc_broken[i + K - 1] - acc_broken[i - 1];
        ans = min(ans, cur);
    }
    printf("%d\n", ans);
    return 0;
}

File: Part1_강의자료/Ch08_투포인터/문제별코드/14465_소가길을건너간이유5/14465_prefix_sum.py
N, K, B = map(int, input().split())
is_broken = [0] * (N + 1)
for _ in range(B):
    is_broken[int(input())] = 1

acc_broken = [0] * (N + 1)
for i in range(1, N + 1):
    acc_broken[i] = acc_broken[i - 1] + is_broken[i]

ans = K
for i in range(1, N - K + 2):
    ans = min(ans, acc_broken[i + K - 1] - acc_broken[i - 1])
print(ans)
